pkg/controllers/override/imageparse.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2023 The KubeAdmiral Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package override
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/distribution/reference"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	utilreference "github.com/kubewharf/kubeadmiral/pkg/lifted/distribution/reference"
+)
+
+type Image struct {
+	registry   string
+	repository string
+	tag        string
+	digest     string
+}
+
+func (i *Image) OperateRegistry(operator, value string) {
+	switch operator {
+	case OperatorAdd:
+		i.registry += value
+	case OperatorReplace:
+		i.registry = value
+	case OperatorRemove:
+		i.registry = ""
+	}
+}
+
+func (i *Image) OperateRepository(operator, value string) {
+	switch operator {
+	case OperatorAdd:
+		i.repository += value
+	case OperatorReplace:
+		i.repository = value
+	case OperatorRemove:
+		i.repository = ""
+	}
+}
+
+func (i *Image) OperateTag(operator, value string) {
+	var newTag string
+	switch operator {
+	case OperatorAdd:
+		newTag = i.tag + value
+	case OperatorReplace:
+		newTag = value
+	case OperatorRemove:
+		newTag = ""
+	}
+
+	if operator != OperatorRemove && !utilreference.AnchoredTagRegexp.MatchString(newTag) {
+		return
```

Comment:
Should we return an error?

Reply:
Yes. We should return an error and propagate to the user in the form of an event.

Reply:
I think an error should be returned here, otherwise the user will not be able to understand why the override does not take effect. The code here has been updated.